plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueIcebergTableOperations.java

```diff
@@ -102,7 +102,9 @@ protected String getRefreshedLocation(boolean invalidateCaches)
     protected void commitNewTable(TableMetadata metadata)
     {
         verify(version == -1, "commitNewTable called on a table which already exists");
-        String newMetadataLocation = writeNewMetadata(metadata, 0);
+        String newMetadataLocation = Optional.ofNullable(metadata.metadataFileLocation())
+                .orElse(writeNewMetadata(metadata, 0));
```

Comment:
is `commitNewTable` being  invoked on an existing (not new) table now?

Reply:
`commitNewTable` will be invoked for new table creation using CREATE statement and for registering an existing table using the `register_table` procedure.

`commitToExistingTable` is the method that gets invoked while altering/inserting into an existing table.

Reply:
> `commitNewTable` will be invoked for new table creation using CREATE statement and for registering an existing table using the `register_table` procedure.

`register_table` procedure registers an existing table (not new). it shouldn't go this path

`commitToExistingTable` isn't good either, since we're not committing anything.

just thinking -- do we need to go through table operations at all?
cc @alexjo2144 @findinpath

Reply:
I'd be in favour of adding an extra operation in `AbstractIcebergTableOperations` for dealing with the registration of tables and not adding more complexity for the `commitNewTable` operation.

@findepi we could use a different hierarchy of classes to deal with hive/glue for this specific use case. I'm not sure whether the new boilerplate classes pay off for a reduced code complexity.

Reply:
Looks like we can revert this line now